src/ppl/src/IOPlacer.cpp

```diff
@@ -755,7 +755,14 @@ bool IOPlacer::assignPinsToSections(int assigned_pins_count)
 
   total_pins_assigned += assigned_pins_count;
 
-  if (total_pins_assigned == net->numIOPins()) {
+  if (total_pins_assigned > net->numIOPins()) {
+    logger_->error(
+        PPL,
+        13,
+        "Internal error, placed more pins than exist ({} out of {}).",
+        total_pins_assigned,
+        net->numIOPins());
+  } else if (total_pins_assigned == net->numIOPins()) {
```

Comment:
Minor nit: I think it makes sense to keep this block and the following as two seperate ifs. Totally up to you if you want to change it.

```c++
  if (total_pins_assigned > net->numIOPins()) {
    logger_->error(
        PPL,
        13,
        "Internal error, placed more pins than exist ({} out of {}).",
        total_pins_assigned,
        net->numIOPins());
  } 

  if (total_pins_assigned == net->numIOPins()) {
  ...
```